frontend/src/metabase-lib/types.ts

```diff
@@ -144,13 +150,10 @@ export type ExpressionArg =
   | ColumnMetadata
   | Clause;
 
-// ExternalOp is a JS-friendly representation of a filter clause or aggregation clause.
-declare const ExternalOp: unique symbol;
 export type ExternalOp = {
-  _opaque: typeof ExternalOp;
   operator: string;
   options: Record<string, unknown>;
-  args: ExpressionArg[];
+  args: [ColumnMetadata, ...ExpressionArg[]];
```

Comment:
`ExternalOp` always has `ColumnMetadata` as the first `args` element. That should let us avoid extra type checks / type guards in filter code